api/src/components/publication/controller.ts

```diff
@@ -224,7 +220,13 @@ export const update = async (
             });
         }
 
-        const updatedPublication = await publicationService.update(event.pathParameters.id, event.body);
+        const updateContent = {
+            ...event.body,
+            ...(event.body.content && { content: helpers.getSafeHTML(event.body.content) }),
+            ...(event.body.topics && { topics: { set: event.body.topics.map((topicId) => ({ id: topicId })) } })
+        } as I.UpdatePublicationData;
```

Comment:
if you move the topics ids mapping inside the update service, then you don't need to cast as I.UpdatePublicationData anymore.

Reply:
Agreed, moved the mapping to the service and removed this cast.